Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.Vessel.swift

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extension AddCustomDynamicEntityDataSourceView {
+    /// A marine vessel that can be decoded from the vessel JSON.
+    struct Vessel {
+        /// The location of the vessel.
+        let geometry: (x: Double, y: Double)
+        /// The attributes of the vessel.
+        let attributes: [String: Any]
+    }
+}
+extension AddCustomDynamicEntityDataSourceView.Vessel: Decodable {
+    private enum CodingKeys: CodingKey {
```

Comment:
To my understanding an enum's name should be singular, option set name can be plural?

Reply:
The name comes directly from [apple doc](https://developer.apple.com/documentation/foundation/archives_and_serialization/encoding_and_decoding_custom_types):

> Codable types can declare a special nested enumeration named CodingKeys that conforms to the [CodingKey](https://developer.apple.com/documentation/swift/codingkey) protocol.

Reply:
This enum is different from other enums in that all cases will be used. There may not be a corresponding value for them in the decoded/encoded data, but these are all the coding keys, so a plural makes sense whereas it doesn't for most other enums.

Reply:
Dang, I forgot the details of doing custom coding!